torch/csrc/TypeInfo.cpp

```diff
@@ -176,12 +165,54 @@ static PyObject* THPFInfo_tiny(THPFInfo* self, void*) {
   });
 }
 
+static PyObject* THPFInfo_resolution(THPFInfo* self, void*) {
+  return AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(at::kHalf, self->type, "digits10", [] {
+    return PyFloat_FromDouble(
+        std::pow(10, -std::numeric_limits<at::scalar_value_type<scalar_t>::type>::digits10));
+  });
+}
+
+PyObject* THPFInfo_str(THPFInfo* self) {
+  auto type = self->type;
+  std::string primary_name, legacy_name;
+  std::tie(primary_name, legacy_name) = torch::utils::getDtypeNames(type);
+  std::ostringstream oss;
+
+  if (type == at::ScalarType::BFloat16) {
+    oss << "finfo(dtype=" << primary_name << ")";
+    return THPUtils_packString(oss.str().c_str());
+  }
```

Comment:
By the way, `std::numeric_limits` seem does not be well adapted to `BFloat16` type. Calling it directly in the current version would cause a Python crash and the error was just like:
```
>>> torch.finfo(torch.bfloat16).max
terminate called after throwing an instance of 'c10::Error'
  what():  "max" not implemented for 'BFloat16
```
So I wrote this logic. Make it only output the `dtype` info, but not `max`/`min`/`tiny` or others when the input is a `BFloat16` type. Hope I haven't done anything redundant.

Reply:
It's true you can't use std::numeric_limits, but can you use the values from https://github.com/pytorch/pytorch/blob/1a0b95e7e46a056546a49f7f8b40aaf6ffa11dde/c10/util/BFloat16-inl.h#L234?

Reply:
Thanks for reminding me, I have made some improvements and the methods of finfo can successfully adapt to `BFloat16` type now!
```
>>> torch.finfo(torch.bfloat16).max
3.3895313892515355e+38
>>> torch.finfo(torch.bfloat16).min
-3.3895313892515355e+38
>>> torch.finfo(torch.bfloat16).bits
16
>>> torch.finfo(torch.bfloat16).dtype
'bfloat16'
>>> torch.finfo(torch.bfloat16).eps
0.0078125
>>> torch.finfo(torch.bfloat16).tiny
1.1754943508222875e-38
>>> torch.finfo(torch.bfloat16).resolution
0.01
>>> torch.finfo(torch.bfloat16)
finfo(resolution=0.01, min=-3.38953e+38, max=3.38953e+38, eps=0.0078125, tiny=1.17549e-38, dtype=bfloat16)
>>>       
```